tests/sentry/api/endpoints/test_project_details.py

```diff
@@ -251,120 +252,145 @@ class ProjectUpdateTestTokenAuthenticated(APITestCase):
         self.create_member(
             user=self.user,
             organization=self.project.organization,
-            teams=[self.project.teams.first()],
             role="member",
         )
+        self.create_team_membership(user=self.user, team=self.team, role="admin")
 
-        token = ApiToken.objects.create(user=self.user, scope_list=["project:write"])
-        authorization = f"Bearer {token.token}"
+        for scope in ["project:write", "project:admin"]:
+            authorization = self.create_bearer_token(self.user, [scope])
 
-        data = {"platform": "rust"}
+            response = self.client.put(
+                self.url, format="json", HTTP_AUTHORIZATION=authorization, data=self.data
+            )
+            assert response.status_code == 200, response.content
 
-        url = reverse(
-            "sentry-api-0-project-details",
-            kwargs={
-                "organization_slug": self.project.organization.slug,
-                "project_slug": self.project.slug,
-            },
+    def test_manager_update_allowed_with_correct_token_scope(self):
+        self.create_member(
+            user=self.user,
+            organization=self.project.organization,
+            teams=[self.team],
+            role="manager",
         )
-        response = self.client.put(url, format="json", HTTP_AUTHORIZATION=authorization, data=data)
-        assert response.status_code == 403, response.content
 
-    def test_admin_updates_allowed_with_correct_token_scope(self):
+        for scope in ["project:write", "project:admin"]:
+            authorization = self.create_bearer_token(self.user, [scope])
+
+            response = self.client.put(
+                self.url, format="json", HTTP_AUTHORIZATION=authorization, data=self.data
+            )
+            assert response.status_code == 200, response.content
+
+    def test_owner_update_allowed_with_correct_token_scope(self):
         self.create_member(
             user=self.user,
             organization=self.project.organization,
-            teams=[self.project.teams.first()],
-            role="admin",
+            teams=[self.team],
+            role="owner",
         )
 
-        token = ApiToken.objects.create(user=self.user, scope_list=["project:write"])
-        authorization = f"Bearer {token.token}"
+        for scope in ["project:write", "project:admin"]:
+            authorization = self.create_bearer_token(self.user, [scope])
 
-        data = {"platform": "rust"}
+            response = self.client.put(
+                self.url, format="json", HTTP_AUTHORIZATION=authorization, data=self.data
+            )
+            assert response.status_code == 200, response.content
 
-        url = reverse(
-            "sentry-api-0-project-details",
-            kwargs={
-                "organization_slug": self.project.organization.slug,
-                "project_slug": self.project.slug,
-            },
+    def test_member_update_denied_with_token(self):
```

Comment:
Are members supposed to be denied even with a token with sufficient scopes? Does this imply that access to our API's is dictated by a combination of token + user info (like role etc.)

If so, does that mean that the same token presented by 2 different users could result in two different outcomes? Generally bearer tokens imply that whoever "bears" the token has the privileges that it grants, do ours not work this way?

Reply:
With user tokens, what happens is we take the intersection of the token's scopes and the user's scopes in the org. If any scope in the intersection can access the endpoint, we let the request through.

The same token cannot be presented by 2 different users because user tokens are tied to a specific user. The backend will know which user the token is from

Reply:
I'll change the test to use `project:read` because I think that makes more sense and add a comment explaining why it failed